src/pip/_internal/utils/misc.py

```diff
@@ -479,9 +478,10 @@ def unzip_file(filename, location, flatten=True):
                 ensure_dir(fn)
             else:
                 ensure_dir(dir)
-                fp = open(fn, 'wb')
+                fp = zip.open(name)
```

Comment:
I would add a code comment saying why this approach is being used (what's in your PR, basically).

Reply:
Okay; I didn't bother with a comment because this PR basically brings `unzip_file` in line with `untar_file` below it (i.e. they both now grab a file-like-object representing an archive member and copy it in chunks with copyfileobj), which doesn't have such commentary.

Still, `TarFile` doesn't give people the (rather dangerous) option of just reading an entire file into memory with a single method, so I could see a comment being warranted to warn anybody thinking about changing it back!

Reply:
Okay, thanks. Then perhaps it’s not necessary. (It could be an extremely brief comment though.)

Reply:
Is that too long?

Reply:
If it were me, I'd do it a bit shorter, e.g. "Don't use read() to avoid allocating a potentially large amount of memory." The NEWS item also seems a bit long (normally they're only one or two lines), but I'm less sure of the style there. Maybe the extra description would be more helpful to people.

Reply:
Okay, I'll see what I can do (I tend to be overly verbose in docs :)

Reply:
Single line NEWS entries are preferred.v

It makes it easier to skim through and get a summary IMO.

Reply:
Sounds good to me!